src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -280,7 +291,11 @@ cpdef object compute_network(
 
     cdef int maxreachlen = max(reach_sizes)
     buf = np.empty((maxreachlen, buf_cols), dtype='float32')
-    out_buf = np.empty((maxreachlen, 3), dtype='float32')
+    
+    if return_courant:
+        out_buf = np.empty((maxreachlen, 6), dtype='float32')
+    else:
+        out_buf = np.empty((maxreachlen, 3), dtype='float32') 
```

Comment:
Conditional sizing of `out_buf`. If Courant metrics are requested, the size of `out_buf` needs to be 3x6.